[JoltPhysics] Allow option to compile with rtti enabled, or make it default

### Is your feature request related to a problem? Please describe.

Yes, I am currently unable to compile my project with Jolt in its current state.

### Proposed solution

See discussion with Jolt's owner [here](https://github.com/jrouwe/JoltPhysics/discussions/1410). His suggestion was to go [here](https://github.com/microsoft/vcpkg/blob/ff8d715b6dfcc192f4930e8e696afe6a2e0e85c0/ports/joltphysics/portfile.cmake#L13-L17) and "Just add something along the lines rtti_enabled CPP_RTTI_ENABLED to that code block." Alternatively the default could be changed to enable rtti.

### Describe alternatives you've considered

No other alternatives while still using vcpkg

### Additional context

Copy pasted from the thread with the owner:
Question: Hello, my project compiles fine when running the sample Hello World file, but I'm currently in the process of implementing a debug renderer and I'm getting the following error:
```
[build] Undefined symbols for architecture arm64:
[build]   "typeinfo for JPH::DebugRenderer", referenced from:
[build]       typeinfo for PhysicsDebugRenderer in libphysics.a[3](physicsDebugRenderer.cpp.o)
[build] ld: symbol(s) not found for architecture arm64
```
I'm using macOS Sonoma 14.4.1, in a VS Code project. Jolt is integrated via vcpkg/CMake with vcpkg install joltphysics[core,debugrenderer,profiler], so the debug renderer should be included there. I ran nm -gC path/to/libJolt.a | grep DebugRenderer and there were a lot of finds, maybe there's somehow no definition for it though. I'm not positive where to go from here. As mentioned, the default parts of the library are building and linking just fine.

Answer: Jolt by default compiles without C++ RTTI, your project is compiling with RTTI. Either turn it off for your own project or turn it on for Jolt using the CPP_RTTI_ENABLED cmake option.
I'm not the maintainer of the vcpkg, but looking at the vcpkg repository:

https://github.com/microsoft/vcpkg/blob/ff8d715b6dfcc192f4930e8e696afe6a2e0e85c0/ports/joltphysics/portfile.cmake#L13-L17

it is not exposed.

Exposing it should be trivial though. Just add something along the lines rtti_enabled CPP_RTTI_ENABLED to that code block and send a PR to microsoft.

--- a/ports/joltphysics/portfile.cmake
+++ b/ports/joltphysics/portfile.cmake
@@ -14,6 +14,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
         debugrenderer       DEBUG_RENDERER_IN_DEBUG_AND_RELEASE
         profiler            PROFILER_IN_DEBUG_AND_RELEASE
+        rtti                CPP_RTTI_ENABLED
 )
 
 vcpkg_cmake_configure(

--- a/ports/joltphysics/vcpkg.json
+++ b/ports/joltphysics/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "joltphysics",
   "version": "5.2.0",
+  "port-version": 1,
   "description": "A multi core friendly rigid body physics and collision detection library suitable for games and VR applications",
   "homepage": "https://github.com/jrouwe/JoltPhysics",
   "license": "MIT",
@@ -20,6 +21,9 @@
     },
     "profiler": {
       "description": "Enable the profiler in Debug and Release builds"
+    },
+    "rtti": {
+      "description": "Enable C++ RTTI"
     }
   }
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3854,7 +3854,7 @@
     },
     "joltphysics": {
       "baseline": "5.2.0",
-      "port-version": 0
+      "port-version": 1
     },
     "josuttis-jthread": {
       "baseline": "2020-07-21",

--- a/versions/j-/joltphysics.json
+++ b/versions/j-/joltphysics.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "35a645135c8c7b4e83a7f3e3d5cb62ea8cb094e4",
+      "version": "5.2.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "60a05c8f3d54f60886a116a6f03cbbe7f76f5b56",
       "version": "5.2.0",
       "port-version": 0

